<div class="col-md-3">
    <div class="box box-solid">
        <div class="box-body">
            <h4 class="">Structured Database</h4>
            <p>
                The annotation of ARGs is structured into three different levels: Antibiotic Class, Antibiotic Resistance Gene, and Antibiotic Resitance Mechanism.
            </p>
            <!-- <div class="col-md-12 col-md-offset-0">
                    <img src="assets/images/main.svg" alt="">
                </div> -->
        </div>
    </div>
</div>

<div class="col-md-3">
    <div class="box box-solid">
        <div class="box-body">
            <h4 class="">Mobile Genetic Elements (MGEs)</h4>
            <p>
                ARG-miner provides evidence of the ARGs being transferred by <strong>M</strong>obile <strong>G</strong>enetic
                <strong>E</strong>lements (MGEs) such as Plasmids, Viruses and Phages.
            </p>
            <p>
                This information enables the identification of ARGs that are critical for propagation and disemination to different environments. This is crucial for the interpretation of risk assessments studies.
            </p>
        </div>
    </div>
</div>
<div class=" col-md-3">
    <div class="box box-solid">
        <div class="box-body">
            <h4 class="">Pathogen-carried ARGs</h4>
            <p>
                ARG-miner also provides information about ARGs being carried by bacterial genomes and out of those, how many genomes are pathogens. This is useful for identifying chromosomal ARGs that are hosted by pathogen bacteria.
            </p>
        </div>
    </div>
</div>

<div class="col-md-3">
    <div class="box box-solid">
        <div class="box-body">
            <h4 class="">Metagenomics Analysis</h4>
            <p>
                ARG-miner database is open sourced and free to download it is currently used by our machine learning metagenomics annotation <strong><a href="http://bench.cs.vt.edu/deeparg">DeepARG</a></strong> model.
            </p>
        </div>
    </div>
</div>


<div class="col-md-12 ">
    <div class="box box-solid">
        <div class="box-body">
            <h4 class="">Crowdsourced Annotation</h4>
            <p>
                Its well known that current ARG databases contain annotation with mistakes that are very easily to propagate. By using crowdsourcing, the ARG annotation is made by a broad community instead of individual labs.
            </p>

            <p>
                ARG-miner is an open, cooperative platform. This means, anyone from anywhere can contribute to the curation of ARGs. You don't need to be a skilled expert to contribute!.
            </p>

            <h5>How to validate a gene that has been reviewed?</h5>

            <p> To validate annotations a scoring function is designed using curations carried out by experts. This function is optimized by computational techniques that maximize the probability of right annotations.
            </p>

            <hr>
            <h4>Reference:</h4>
            <p>
                Paper under submission
            </p>
            <hr>
            <h4>Contact us</h4>
            email: lqzhang@vt.edu <br> webpage: <a href="https://bioinformatics.cs.vt.edu/zhanglab/">https://bioinformatics.cs.vt.edu/zhanglab/</a>
            <hr>
            <h4>Open Source</h4>
            ARG-miner source code can be found at the git repository <code> https://github.com/gaarangoa/ARG-inspect.git
      </code>
            <hr>
            <h4>API</h4>
            ARG-miner data can be programatically acccessed through its restAPI (documentation under construction).
        </div>
    </div>






</div>