<div class="">

    <div *ngIf="!render" class="col-md-4 col-md-offset-4">
        <div class="col-sm-4 col-sm-offset-4">
            <object>
            <embed src="assets/images/loading_1.svg" />
          </object>
        </div>
    </div>



    <div *ngIf="render">
        <!--Left panel-->
        <div class="col-sm-3 ">
            <!-- <div class=""> -->
            <div *ngIf="randomARG.status==true" class="">
                <!-- <div class="col-md-12"> -->
                <p-growl class="user_growl" [(value)]="notification"></p-growl>
                <div class="box box-solid box-primary">
                    <div class="box-header with-border">



                        <i class="fa fa-gears"></i>
                        <!-- <p *ngIf="loading"><md-progress-bar mode="indeterminate"></md-progress-bar></p>  -->
                        <h3 class="box-title"><strong>Current Annotation </strong> </h3>

                    </div>

                    <!-- /.box-header -->

                    <!-- <tour-step class="intro-tour-hint-wrapper" selector="step1" order="1" position="right">
                        <div class="box box-step">
                            <div class="box-header">
                                <h3 class="box-title">
                                    ARG you have to curate
                                </h3>
                            </div>
                            <div class="box-body">
                                This panel shows you the basic information about the antibiotic reistance gene you need to inspect. Note this information may contain errors. Therefore, by comparing this information with the known ARGs databases you will be able to correct or accept
                                the current annotation.
                            </div>
                        </div>
                    </tour-step> -->

                    <div class="box-body">

                        <dl id="step1" class="dl">
                            <!-- <p> -->
                            <strong>Query gene id:</strong>
                            <span class="">{{ randomARG.entry.gene_id }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>Antibiotic Class:</strong>
                            <span class="">{{ randomARG.entry.type }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>Gene Name:</strong>
                            <span class="">{{ randomARG.entry.subtype }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>Database:</strong>
                            <span class="">{{ randomARG.entry.database }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong data-intro="step1">Inspected:</strong>
                            <span class="">{{ randomARG.inspected.length }}</span>
                            <br>
                            <!-- </p> -->
                            <!-- <dt>Score:</dt> -->
                            <!-- <dd><span class="badge bg-blue">{{ randomARG.entry.score }}</span></dd> -->
                        </dl>

                        <p class="small">*ARG: <u><strong>A</strong>ntibiotic <strong>R</strong>esistance <strong>G</strong>ene</u></p>


                        <hr>
                        <!-- <h4>Tutorial</h4> -->

                        <div class="text-center">
                            <a (click)="startTour()" class="badge bg-blue btn-xs"><span class=""> <strong> <u>Tour </u></strong> </span></a>
                            <a (click)="openInstructions()" class="badge bg-red btn-xs"><span class=""> <strong> <u>Instructions</u></strong> </span></a>
                        </div>
                        <br>
                        <!-- <hr> -->

                        <!-- <h4>Filter options</h4> -->

                        <!-- <tour-step class="intro-tour-hint-wrapper" selector="step2" order="2">
                            <div class="col-md-12">
                                <div class="box box-primary">
                                    <div class="box-header">
                                        <h3 class="box-title">
                                            <strong>I am a New user</strong>
                                        </h3>
                                    </div>
                                    <div class="box-body">
                                        If you are a new user, activate the new user button. You will have to complete 20 entries that will help you to get familiar with the system. I will also score your results based on the real annotation.
                                    </div>
                                    <div class="box-footer">
                                       
                                    </div>
                                </div>
                            </div>
                            
                        </tour-step> -->

                        <div id="step3">
                            <md-slide-toggle (change)="trainingARGEvent($event)" [checked]="trainingARGFlag"><strong>New User</strong></md-slide-toggle>
                            <p class="small">Enable this option if this is the first time you enter the website.</p>
                        </div>
                        <!-- <hr> -->
                        <div id="step4">
                            <md-slide-toggle (change)="selectConflictedArgEvent($event)" [checked]="conflictedArgSubtypeFlag"><strong>Priority ARGs</strong></md-slide-toggle>
                            <p class="small">This option selects ARGs with high priority of curation.</p>
                        </div>
                        <hr>



                        <div *ngIf="loading==false && conflictedArgSubtypeFlag">

                            <h4> Conflicted ARGs </h4>
                            <p class="small">This section contains the list of ARGs under the category {{ conflictingARGSubtype.subtype }} that present inconsistences respect its ARG class annotation.</p>
                            <!-- <div class="box-body"> -->
                            <p> <strong>Gene:</strong> <mark>{{ conflictingARGSubtype.subtype }}</mark></p>
                            <div *ngFor="let item of conflictingARGSubtype.conflict">
                                <strong>{{ item.class }}</strong> {{ item.genes.length }}
                            </div>
                            <hr>
                            <button (click)="nextGene()" class="" color="primary" md-raised-button>Other ARG Group</button>
                            <button (click)="nextGeneConflictList()" class="" color="warn" md-raised-button>Next ARG</button>
                            <!-- </div> -->
                        </div>

                        <div *ngIf="conflictedArgSubtypeFlag==false && trainingARGFlag==false" class="col-md-12 text-center">
                            <button (click)="nextGene()" class="" color="primary" md-raised-button>Random ARG</button>
                        </div>

                        <div *ngIf="trainingARGFlag && conflictedArgSubtypeFlag==false" class="col-md-12 text-center">
                            <button (click)="trainingARGNextGene()" class="" color="primary" md-raised-button>Next ARG</button>
                        </div>


                        <!-- <div class="col-md-12 text-right">
                            <button (click)="nextGene()" class="" color="red" md-raised-button>Random</button>
                        </div> -->


                    </div>

                    <!-- /.box-body -->
                    <!-- </div> -->
                    <!-- /.box -->
                </div>
            </div>

            <!-- <div class="col-md-12 box box-solid ">

                <h4>Note about BitScores</h4>
                <p class="small">A bitscore of 50 is considered significant if the alignment covers at least 90% of the reference sequence</p>
                <p class="small">Bit-score does not depend on database size. The bit-score gives the same value for hits in databases of different sizes and hence can be used for searching in an constantly increasing database.</p>
                <p class="small"> The higher the bitscore, the better the quality of the sequence alignment</p>

            </div> -->


            <md-expansion-panel id="step12" class="">
                <md-expansion-panel-header>
                    <md-panel-title>
                        <h4>Micro-Tasks</h4>
                    </md-panel-title>
                </md-expansion-panel-header>
                <h4>ARG identification</h4>
                <p>Please fill up the Antibiotic resistance class, group and mechanism (if any) of the gene based on the observation from the different resources.</p>

                <p class="small">The <strong>Blast alignment</strong> sections associate the queried gene to its closest antibiotic resistance gene from curated databases. From this section you need to look at the scores (Bitscore, evalue, identity and coverage). High
                    identity and coverage represent close related sequences. Statistically significant alignments don't necesarly need high identity percentages. Therefore, pay attention to the Bitscores and Evalues. <strong>Note</strong>: A bitscore
                    greater than 50 is considered significant if it spans at least 90% coverage.
                </p>


                <hr>

                <h4> Mobile Genetic Elements</h4>

                <p><strong>Horizontal Gene Transfer</strong></p>
                <p>Please check if there is any evidence that suggests the gene has been observed in a plasmid, prophage or virus.</p>
                <p class="small">To complete this task please check the Plasmid, Virus, or Prophages sections. If there is not any section with plasmid, prophage or virus tags it means that there is not evidence that the gene is transfered by any of those mechanisms.
                </p>
                <p><strong>Pathogen Genomes</strong></p>
                <p>Please take a look at the pathogen genomes panel under the horizontal gene transfer section. </p>
                <p class="small">This panel will show you the association of the queried gene against a set of ~100,000 bacterial genomes. You will get information about the association of the gene to pathogens (as percentages) as well as their antimicrobial phenotype
                    (resistant, susceptible, intermediate). Rate your findigns according to the provided evidence.

                </p>

                <hr>
                <h4>Review and submit</h4>
                <p>This panel contains your observations for the queried gene.</p>
                <p><strong>Verify your observations</strong></p>
                <p>Please, make sure the values in the fields (ARG class, ARG group and ARG mechanism) correspond to what you have observed.</p>
                <p><strong>Rate your expertise</strong></p>
                <p>Please in a scale from 1 to 5 rate your confidence or expertice level, where 1 depicts a non expert evaluator, and 5 if you have experience working with Antibiotic Resistance Genes.
                    <p>
                        <p><strong>Rate your annotation confidence</strong></p>
                        <p>Please in a scale from 1 to 5, rate your confidence in the curation of the queried gene where 1 means that there is a lack of evidence about the gene and 5 the data correlates with your observations.</p>
            </md-expansion-panel>
            <br>
            <md-expansion-panel>
                <md-expansion-panel-header>
                    <md-panel-title>
                        <h4>Similarity to known ARGs</h4>
                    </md-panel-title>
                </md-expansion-panel-header>
                <p>
                    To curate a gene as an ARG, it is necesary to validate its evidence. One way to do it is by looking at the information provided from curated databases such as the <a href="">CARD</a> database. Thus, the entry {{ randomARG.entry.gene_id
                    }} was compared against the databases to reinforce, reject or modify the <strong>current annotation.</strong>
                </p>
                <p>
                    For this comparison, the gene {{ randomARG.entry.gene_id }} was aligned to the genes from the databases using blast and its best hit was extracted. For each database, a panel shows the quality of the alignment and the information regarding that best hit.
                </p>

                <p> <span class="label bg-blue">Tip</span> To identify the
                    <strong> class of antibiotic </strong> consider looking at <u>middle</u> to <u>high</u> bitscore, evalue and coverage. However, to identify <strong> antibiotic resistance names </strong>, we would recomend to look at the panels where
                    the sequence similarity is <u>high</u> with a <u>very strong</u> coverage.
                </p>

                <p class="">*Each color represents a different scoring scale</p>

            </md-expansion-panel>
            <br>
            <md-expansion-panel>
                <md-expansion-panel-header>
                    <md-panel-title>
                        <h4>MGEs and pathogens</h4>
                    </md-panel-title>
                </md-expansion-panel-header>
                <p>
                    Mobile Genetic Elements or MGEs are segments that encode proteins that mediates the movment of DNA within genomes. Plasmids are collection of stable funcional genetic elements. Plasmids can often be transferred to another cell. Bacteriophages can also
                    transfer genetic material by accidentally package segments of DNA to inject into their host.
                </p>
                <p>
                    To identify if the ARG is carried by plasmids, phages or viruses, the gene {{ randomARG.entry.gene_id }} was aligned to the <a href="">ACLAME</a> database. Scores provide insight into the presence of this gene into the MGE.
                </p>
                <p>
                    Pathogens are invaders that attack their hosts. The most familiar are viruses and bacteria and they can cause a wide range of infections and diseases. To identify if the gene {{ randomARG.entry.gene_id }} is hosted in a pathogen genome, the gene was matched
                    to the <a href="">PATRIC</a> database. Then, we were able to identify the number of pathogen genomes that carry the {{ randomARG.entry.gene_id }} gene.
                </p>
                <p class="small">Each color represents a different scoring scale</p>
            </md-expansion-panel>
            <!-- </div> -->
        </div>


        <div class="col-sm-6 col-sm-offset-0">

            <div id="step5" class="">
                <div class="box box-solid ">
                    <div class="box-body">
                        <md-input-container class='col-sm-12 col-sm-offset-0'>
                            <input #keyword type="search" (keyup.enter)="search(keyword.value)" mdInput placeholder="Search term " value="" />
                        </md-input-container>
                    </div>
                </div>
            </div>



            <div *ngIf="loading==true" class="col-md-4 col-md-offset-4">
                <div class="col-sm-8 col-sm-offset-2">
                    <object>
                <embed src="assets/images/loading_1.svg" />
              </object>
                </div>
            </div>


            <div *ngIf="randomARG.status==true && loading==false">

                <hr>


                <md-expansion-panel class="box box-solid">

                    <md-expansion-panel-header>
                        <md-panel-title>
                            Tips and Tricks
                        </md-panel-title>
                        <md-panel-description>
                            To get a successful annotation take into account the next tips:
                        </md-panel-description>
                    </md-expansion-panel-header>

                    <div class="box-body">
                        <ul>
                            <li>
                                <strong>Gene Name:</strong> To validate the gene name take a look at the panels with <span class="label bg-blue"> strong </span> and <span class="label bg-black">very strong</span> evidence. If several panels have strong
                                evidence, select the gene name in the panel with the highest bitscore. If none of the panels have strong evidence, then, keep the <u>gene name</u> as in the <strong> current annotation</strong>.
                            </li>
                            <li>
                                <strong>Antibiotic Class:</strong> To validate the antibiotic class take a look at the descriptions from the databases. There is not restrction about evidence. But it is expected to see a consensus between the different
                                databases.
                            </li>
                            <li>
                                <strong>Antibiotic Mechanism:</strong> Check the AMR profile and Metadata (if available) to find the antibiotic mechanism. If the evidence from the <strong>MEGARes</strong> database is <span class="label bg-green"> middle </span>,
                                <span class="label bg-blue"> strong </span> or <span class="label bg-black"> very strong </span>, it is safe to set this mechanism. Other case, seek for consensus from the different panels.
                            </li>
                        </ul>
                    </div>
                </md-expansion-panel>

                <br>
                <h4>Similarity to known ARGs</h4>
                <!-- <md-tab-group (selectChange)="tabsEvent($event)"> -->
                <!-- <md-tab  label="METADATA"> -->


                <div *ngFor="let bh of randomARG.besthit.alignments">
                    <!-- <p> {{ bh.best_hit_database }} </p>  -->

                    <div id="step8" *ngIf="bh.best_hit_database=='ARDB'">
                        <besthit-metadata-ardb> </besthit-metadata-ardb>
                    </div>
                    <div id="step7" *ngIf="bh.best_hit_database=='CARD'">
                        <besthit-metadata-card> </besthit-metadata-card>
                    </div>
                </div>
                <!-- </md-tab> -->

                <!-- OTHER DATABASES  -->
                <div id="step9">
                    <app-generic></app-generic>
                </div>

                <h4 *ngIf="randomARG.entry.database=='UNIPROT'">Metadata</h4>

                <!--METADATA SECTION-->
                <div id="step6">
                    <classify-metadata-uniprot *ngIf="randomARG.entry.database=='UNIPROT'"> </classify-metadata-uniprot>
                </div>


                <classify-metadata-card> </classify-metadata-card>


                <!-- </md-tab> -->

                <!-- <md-tab  label="BEST HIT"> -->
                <hr *ngIf="randomARG.entry.database=='UNIPROT'">



                <h4>Mobile Genetic Elements (MGEs)</h4>



                <!-- <md-tab  label="HORIZONTAL GENE TRANSFER"> -->
                <!-- <div *ngIf="drawGenomes"> -->
                <div id="step10">
                    <genome-metadata *ngIf="randomARG.entry.database=='UNIPROT'"></genome-metadata>
                    <genome-metadata *ngIf="randomARG.entry.database=='CARD'"></genome-metadata>
                    <genome-metadata *ngIf="randomARG.entry.database=='ARDB'"></genome-metadata>
                </div>
                <!-- </div> -->
                <!-- </md-tab> -->

                <!-- </md-tab-group> -->
                <hr>
                <div>
                    <div class="box box-solid ">
                        <div class="box-header with-border">
                            <h3 class="box-title"> <strong>Protein sequence</strong> ({{ randomARG.entry.gene_id }}) </h3>
                        </div>
                        <div class="box-body">
                            <p class="sequence small"> ><strong>{{randomARG.entry.gene_id}}</strong> <br> {{ randomARG.entry.sequence }}</p>
                        </div>
                    </div>
                </div>
                <br><br><br><br>
            </div>
        </div>


        <!-- <div class="col-sm-3"> -->
        <div id="step11" class="curate-nav affix clarify">
            <!-- <aside class="control-sidebar control-sidebar-open"> -->
            <app-curate></app-curate>
            <!-- </aside> -->
        </div>
        <!-- </div> -->
    </div>

    <!--</md-tab>-->

    <!--<md-tab label="Novel ARGs">Content 2</md-tab>-->

    <!--<md-tab label="Potential ARGs">Content 2</md-tab>-->

    <!--</md-tab-group>-->
</div>