<div class="">

    <div *ngIf="!render" class="col-md-4 col-md-offset-4">
        <div class="col-sm-4 col-sm-offset-4">
            <object>
            <embed src="assets/images/loading_1.svg" />
          </object>
        </div>
    </div>


    <div *ngIf="render">
        <!--Left panel-->
        <div class="col-sm-3 ">
            <!-- <div class=""> -->
            <div *ngIf="randomARG.status==true" class="">
                <!-- <div class="col-md-12"> -->
                <p-growl class="user_growl" [(value)]="notification"></p-growl>
                <div class="box box-solid box-primary">
                    <div class="box-header with-border">



                        <i class="fa fa-gears"></i>
                        <!-- <p *ngIf="loading"><md-progress-bar mode="indeterminate"></md-progress-bar></p>  -->
                        <h3 class="box-title"><strong>Current Annotation </strong> </h3>

                    </div>
                    <!-- /.box-header -->
                    <div class="box-body">

                        <dl class="dl">
                            <!-- <p> -->
                            <strong>Query gene id:</strong>
                            <span class="">{{ randomARG.entry.gene_id }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>ARG Class:</strong>
                            <span class="">{{ randomARG.entry.type }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>ARG:</strong>
                            <span class="">{{ randomARG.entry.subtype }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <br>
                            <strong>ARG Database:</strong>
                            <span class="">{{ randomARG.entry.database }}</span>
                            <!-- </p> -->
                            <!-- <p> -->
                            <hr>
                            <strong>Inspected:</strong>
                            <span class="">{{ randomARG.inspected.length }}</span>
                            <!-- </p> -->
                            <!-- <dt>Score:</dt> -->
                            <!-- <dd><span class="badge bg-blue">{{ randomARG.entry.score }}</span></dd> -->

                        </dl>

                        <hr>

                        <h4>Filter options</h4>

                        <md-slide-toggle (change)="selectConflictedArgEvent($event)" [checked]="conflictedArgSubtypeFlag"><strong>Select Conflicting ARGs</strong></md-slide-toggle>
                        <p class="small">This option only takes the ARGs that have two or more annotations. In many cases, these genes have conflicting annotations from the reference database and require special attention. By turning off this filter you will get a random
                            gene from the database.</p>

                        <hr>

                        <md-slide-toggle (change)="trainingARGEvent($event)" [checked]="trainingARGFlag"><strong>Train a New User</strong></md-slide-toggle>
                        <p class="small">This option enables a specific set of ARGs that have been manualy validated to score your performance. </p>
                        <hr>
                        <div *ngIf="loading==false && conflictedArgSubtypeFlag">

                            <h4> Conflicted ARGs </h4>
                            <p class="small">This section contains the list of ARGs under the category {{ conflictingARGSubtype.subtype }} that present inconsistences respect its ARG class annotation.</p>
                            <!-- <div class="box-body"> -->
                            <p> <strong>Gene:</strong> <mark>{{ conflictingARGSubtype.subtype }}</mark></p>
                            <div *ngFor="let item of conflictingARGSubtype.conflict">
                                <strong>{{ item.class }}</strong> {{ item.genes.length }}
                            </div>
                            <hr>
                            <button (click)="nextGene()" class="" color="primary" md-raised-button>Other ARG Group</button>
                            <button (click)="nextGeneConflictList()" class="" color="warn" md-raised-button>Next ARG</button>
                            <!-- </div> -->
                        </div>

                        <div *ngIf="conflictedArgSubtypeFlag==false && trainingARGFlag==false" class="col-md-12 text-center">
                            <button (click)="nextGene()" class="" color="primary" md-raised-button>Random ARG</button>
                        </div>

                        <div *ngIf="trainingARGFlag && conflictedArgSubtypeFlag==false" class="col-md-12 text-center">
                            <button (click)="trainingARGNextGene()" class="" color="primary" md-raised-button>Next ARG</button>
                        </div>


                        <!-- <div class="col-md-12 text-right">
                            <button (click)="nextGene()" class="" color="red" md-raised-button>Random</button>
                        </div> -->


                    </div>

                    <!-- /.box-body -->
                    <!-- </div> -->
                    <!-- /.box -->
                </div>
            </div>

            <div class="col-md-12 box box-solid ">

                <h4>Note about BitScores</h4>
                <p class="small">A bitscore of 50 is considered significant if the alignment covers at least 90% of the reference sequence</p>
                <p class="small">Bit-score does not depend on database size. The bit-score gives the same value for hits in databases of different sizes and hence can be used for searching in an constantly increasing database.</p>
                <p class="small"> The higher the bitscore, the better the quality of the sequence alignment</p>

            </div>

            <div class="col-md-12 box box-solid ">

                <h4><strong>1: ARG identification</strong></h4>
                <p>Please fill up the Antibiotic resistance class, group and mechanism (if any) of the gene based on the observation from the different resources.</p>

                <p>The Blast alignment sections associate the queried gene to its closest antibiotic resistance genes from curated databases. From this section you need to look at the alignment scores (Bitscore, evalue, identity and coverage). High identity
                    and coverage represent close related sequences. Statistically significant alignments don't necesarly need high identity percentages. Therefore, pay attention to the Bitscores and Evalues. <strong>Note</strong>: A bitscore greater than
                    50 is considered significant if it spans at least 90% coverage.
                </p>


                <hr>

                <h4><strong>2: Mobile Genetic Elements</strong></h4>

                <p><strong>Horizontal Gene Transfer</strong></p>
                <p>Please check if there is any evidence that suggests the gene has been observed in a plasmid, prophage or virus.</p>
                <p>To complete this task please check the Horizontal Gene Transfer sections. If there is not any box with plasmid, prophage or virus tags it means to date there is not evidence that the gene is transfered by any of those mechanisms.
                </p>
                <p><strong>Pathogen Genomes</strong></p>
                <p>Please take a look at the pathogen genomes panel under the horizontal gene transfer section. This panel will show you the association of the queried gene against a set of ~100,000 bacterial genomes. You will get information about the association
                    of the gene to pathogens as well as their antimicrobial phenotype (resistant, susceptible, intermediate). Rate your findigns according to the provided evidence.

                </p>

                <hr>
                <h4><strong>3: Review and submit</strong></h4>
                <p>This panel contains your observations for the queried gene.</p>
                <p><strong>Verify your observations</strong></p>
                <p>Please, make sure the values in the fields (ARG class, ARG group and ARG mechanism) correspond to what you have observed.</p>
                <p><strong>Rate your expertise</strong></p>
                <p>Please in a scale from 1 to 5 rate your confidence or expertice level, where 1 depicts a non expert evaluator, and 5 if you have experience working with Antibiotic Resistance Genes.
                    <p>
                        <p><strong>Rate your annotation confidence</strong></p>
                        <p>Please in a scale from 1 to 5, rate your confidence in the curation of the queried gene where 1 means that there is a lack of evidence about the gene and 5 the data correlates with your observations.</p>
            </div>


            <!-- </div> -->
        </div>


        <div class="col-sm-6 col-sm-offset-0">

            <div class="">
                <div class="box box-solid ">
                    <div class="box-body">
                        <md-input-container class='col-sm-12 col-sm-offset-0'>
                            <input #keyword (keyup.enter)="search(keyword.value)" mdInput placeholder="Search term " value="" />
                        </md-input-container>
                    </div>
                </div>
            </div>



            <div *ngIf="loading==true" class="col-md-4 col-md-offset-4">
                <div class="col-sm-8 col-sm-offset-2">
                    <object>
                <embed src="assets/images/loading_1.svg" />
              </object>
                </div>
            </div>

            <div *ngIf="randomARG.status==true && loading==false">
                <!-- <md-tab-group (selectChange)="tabsEvent($event)"> -->
                <!-- <md-tab  label="METADATA"> -->

                <!--METADATA SECTION-->
                <classify-metadata-uniprot *ngIf="randomARG.entry.database=='UNIPROT'"> </classify-metadata-uniprot>
                <classify-metadata-card> </classify-metadata-card>


                <!-- </md-tab> -->

                <!-- <md-tab  label="BEST HIT"> -->
                <hr>
                <div *ngFor="let bh of randomARG.besthit.alignments">
                    <!-- <p> {{ bh.best_hit_database }} </p>  -->
                    <besthit-metadata-ardb *ngIf="bh.best_hit_database=='ARDB'"> </besthit-metadata-ardb>
                    <besthit-metadata-card *ngIf="bh.best_hit_database=='CARD'"> </besthit-metadata-card>
                </div>
                <!-- </md-tab> -->

                <!-- OTHER DATABASES  -->

                <app-generic></app-generic>


                <!-- <md-tab  label="HORIZONTAL GENE TRANSFER"> -->
                <!-- <div *ngIf="drawGenomes"> -->
                <genome-metadata *ngIf="randomARG.entry.database=='UNIPROT'"></genome-metadata>
                <genome-metadata *ngIf="randomARG.entry.database=='CARD'"></genome-metadata>
                <genome-metadata *ngIf="randomARG.entry.database=='ARDB'"></genome-metadata>
                <!-- </div> -->
                <!-- </md-tab> -->

                <!-- </md-tab-group> -->

                <div>
                    <div class="box box-solid ">
                        <div class="box-header with-border">
                            <h3 class="box-title"> <strong>Protein sequence</strong> ({{ randomARG.entry.gene_id }}) </h3>
                        </div>
                        <div class="box-body">
                            <p class="sequence"> ><strong>{{randomARG.entry.gene_id}}</strong> <br> {{ randomARG.entry.sequence }}</p>
                        </div>
                    </div>
                </div>
                <br><br><br><br>
            </div>
        </div>


        <!-- <div class="col-sm-3"> -->
        <div class="curate-nav affix">
            <app-curate></app-curate>
        </div>
        <!-- </div> -->
    </div>

    <!--</md-tab>-->

    <!--<md-tab label="Novel ARGs">Content 2</md-tab>-->

    <!--<md-tab label="Potential ARGs">Content 2</md-tab>-->

    <!--</md-tab-group>-->
</div>